### ISSUE TITLE
Changes compared to each other

### ISSUE THREAD

**Issue by Veranep** [Veranep/TMCI_Kamiel_Baran_Vera#8]
All models use no attention, 10000 pairs from a sample set, and no teacher forcing unless otherwise specified. Furthermore all features are kept the same as in the base model, except for the feature that we are testing.

- Base model:  0.01850195563099788
(['we', 'staan', 'erop'], ['we', 'zijn', 'het', '<EOS>'])
(['dit', 'hadden', 'we', 'niet', 'verwacht'], ['we', 'zijn', 'op', 'op', '<EOS>'])
(['en', 'wij', 'staan', 'in', 'het', 'midden'], ['we', 'zijn', 'de', '<EOS>'])
(['je', 'bent', 'precies', 'wat', 'ik', 'nodig', 'heb'], ['je', 'bent', 'het', '<EOS>'])
(['het', 'spijt', 'me,', 'noel'], ['ik', 'ben', 'me', '<EOS>'])
(['je', 'bent', 'een', 'juweeltje'], ['je', 'bent', 'een', '<EOS>'])
(['graag', 'gedaan,', 'elizabeth'], ['je', 'bent', 'het', '<EOS>'])
(['kan', 'ik', 'aan,', 'geloof', 'me'], ['ik', 'ben', '<EOS>'])
(['wij', 'zijn', 'louis', 'en', 'jessica', 'huang,', 'we', 'wonen', 'in', 'orlando'], ['we', 'zijn', 'op', 'op', '<EOS>'])
(['ik', 'ben', 'het', 'jaar', 'van', 'de', 'hond'], ['ik', 'ben', 'de', 'van', '<EOS>'])

- LogSigmoid: 0.0
(['we', 'staan', 'erop'], ['ik', 'ben', 'ben', '<EOS>'])
(['dit', 'hadden', 'we', 'niet', 'verwacht'], ['ik', 'ben', 'ben', '<EOS>'])
(['en', 'wij', 'staan', 'in', 'het', 'midden'], ['ik', 'ben', 'ben', '<EOS>'])
(['je', 'bent', 'precies', 'wat', 'ik', 'nodig', 'heb'], ['ik', 'ben', 'ben', '<EOS>'])
(['het', 'spijt', 'me,', 'noel'], ['ik', 'ben', 'ben', '<EOS>'])
(['je', 'bent', 'een', 'juweeltje'], ['ik', 'ben', 'ben', '<EOS>'])
(['graag', 'gedaan,', 'elizabeth'], ['ik', 'ben', 'ben', '<EOS>'])
(['kan', 'ik', 'aan,', 'geloof', 'me'], ['ik', 'ben', 'ben', '<EOS>'])
(['wij', 'zijn', 'louis', 'en', 'jessica', 'huang,', 'we', 'wonen', 'in', 'orlando'], ['ik', 'ben', 'ben', '<EOS>'])
(['ik', 'ben', 'het', 'jaar', 'van', 'de', 'hond'], ['ik', 'ben', 'ben', '<EOS>'])

- LeakyReLU: 0.0
(['we', 'staan', 'erop'], ['ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik'])
(['dit', 'hadden', 'we', 'niet', 'verwacht'], ['ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik'])
(['en', 'wij', 'staan', 'in', 'het', 'midden'], ['ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik'])
(['je', 'bent', 'precies', 'wat', 'ik', 'nodig', 'heb'], ['ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik'])
(['het', 'spijt', 'me,', 'noel'], ['ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik'])
(['je', 'bent', 'een', 'juweeltje'], ['ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik'])
(['graag', 'gedaan,', 'elizabeth'], ['ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik'])
(['kan', 'ik', 'aan,', 'geloof', 'me'], ['ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik'])
(['wij', 'zijn', 'louis', 'en', 'jessica', 'huang,', 'we', 'wonen', 'in', 'orlando'], ['ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik'])
(['ik', 'ben', 'het', 'jaar', 'van', 'de', 'hond'], ['ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik', 'ik'])

- Softmax: 0.0
(['we', 'staan', 'erop'], ['slavinnen', 'gevangenenraad', 'dienaar', 'goldman', 'transpireer', 'idina', 'idina', 'gehaald', 'gehaald', 'halvorsen', 'halvorsen', 'suggesties', 'halvorsen', 'halvorsen', 'halvorsen'])
(['dit', 'hadden', 'we', 'niet', 'verwacht'], ['slavinnen', 'aden', 'gevangenenraad', 'aden', 'poolse', 'aden', 'aden', 'beschadigd', 'grotere', 'gehaald', 'zei:', 'gehaald', 'halvorsen', 'halvorsen', 'suggesties'])
(['en', 'wij', 'staan', 'in', 'het', 'midden'], ['slavinnen', 'slavinnen', 'gevangenenraad', 'dienaar', 'smolt', 'boudoir', 'onderneming:', 'versterken', 'dezelfde', 'omatsu', 'basis', 'halvorsen', 'halvorsen', 'halvorsen', 'suggesties'])
(['je', 'bent', 'precies', 'wat', 'ik', 'nodig', 'heb'], ['slavinnen', 'aden', 'gevangenenraad', 'aden', 'poolse', 'aden', 'aden', 'beschadigd', 'grotere', 'gehaald', 'zei:', 'gehaald', 'halvorsen', 'halvorsen', 'suggesties'])
(['het', 'spijt', 'me,', 'noel'], ['ontaard', 'schedel', 'herladen', 'details', 'halvorsen', 'halvorsen', 'halvorsen', 'halvorsen', 'suggesties', 'halvorsen', 'halvorsen', 'halvorsen', 'suggesties', 'halvorsen', 'halvorsen'])
(['je', 'bent', 'een', 'juweeltje'], ['ontaard', 'schedel', 'izar', 'halvorsen', 'halvorsen', 'halvorsen', 'suggesties', 'halvorsen', 'halvorsen', 'halvorsen', 'suggesties', 'halvorsen', 'halvorsen', 'halvorsen', 'suggesties'])
(['graag', 'gedaan,', 'elizabeth'], ['slavinnen', 'aden', 'gevangenenraad', 'goldman', 'strop', 'absoluut', 'halvorsen', 'halvorsen', 'halvorsen', 'suggesties', 'halvorsen', 'halvorsen', 'halvorsen', 'suggesties', 'halvorsen'])
(['kan', 'ik', 'aan,', 'geloof', 'me'], ['slavinnen', 'aden', 'gevangenenraad', 'goldman', 'christus', 'sara', 'tofoekroketten', 'halvorsen', 'halvorsen', 'halvorsen', 'suggesties', 'halvorsen', 'halvorsen', 'halvorsen', 'suggesties'])
(['wij', 'zijn', 'louis', 'en', 'jessica', 'huang,', 'we', 'wonen', 'in', 'orlando'], ['ontaard', 'schedel', 'aden', 'integendeel,', 'aden', 'poolse', 'aden', 'aden', 'beschadigd', 'grotere', 'gehaald', 'zei:', 'gehaald', 'halvorsen', 'halvorsen'])
(['ik', 'ben', 'het', 'jaar', 'van', 'de', 'hond'], ['slavinnen', 'doodgeschoten', 'andrews', 'versterken', 'omatsu', 'basis', 'halvorsen', 'halvorsen', 'halvorsen', 'suggesties', 'halvorsen', 'halvorsen', 'halvorsen', 'suggesties', 'halvorsen'])

- Softmin: 0.0
(['we', 'staan', 'erop'], ['verheffen', 'ravenhurst', 'rinaldi', 'grillig', 'taiko', 'machtiger', 'duitse', 'inspector', 'wieltje', 'slechts', 'kirok', 'bartolome', 'larry', 'bartolome', 'larry'])
(['dit', 'hadden', 'we', 'niet', 'verwacht'], ['ravenhurst', 'dans', 'december', 'hallo', 'hallo', 'affris,', 'lullen', 'landbezitter,', 'verschillende', 'lullen', 'landbezitter,', 'verschillende', 'lullen', 'landbezitter,', 'verschillende'])
(['en', 'wij', 'staan', 'in', 'het', 'midden'], ['raken', 'mordor', 'vrijheid', 'verachtelijk', 'vrijheid', 'vrijheid', 'vrijheid', 'verachtelijk', 'vrijheid', 'vrijheid', 'vrijheid', 'verachtelijk', 'vrijheid', 'vrijheid', 'vrijheid'])
(['je', 'bent', 'precies', 'wat', 'ik', 'nodig', 'heb'], ['vanuit', 'ravenhurst', 'dans', 'december', 'lullen', 'afgepakt', 'vrijheid', 'vrijheid', 'vrijheid', 'verachtelijk', 'vrijheid', 'vrijheid', 'vrijheid', 'verachtelijk', 'vrijheid'])
(['het', 'spijt', 'me,', 'noel'], ['mississippi', 'slagzij', 'britse,', 'gebouwd', 'mordor', 'vrijheid', 'vrijheid', 'vrijheid', 'verachtelijk', 'vrijheid', 'vrijheid', 'vrijheid', 'verachtelijk', 'vrijheid', 'vrijheid'])
(['je', 'bent', 'een', 'juweeltje'], ['slaapt', 'scherpschutter', 'patrouille', 'zinloos', 'bartolome', 'bartolome', 'larry', 'larry', 'sliep', 'lilliput', 'ge', 'machtiger', 'duitse', 'gids,', 'ravenhurst'])
(['graag', 'gedaan,', 'elizabeth'], ['zangeres', 'préfet', 'shogun', 'taiko', 'gearresteerd,', 'bespioneren', 'hoge', 'lullen', 'dieper', 'taiko', 'bloem', 'gearresteerd,', 'lullen', 'schoonmoeder,', 'duitse'])
(['kan', 'ik', 'aan,', 'geloof', 'me'], ['james', 'cool', 'gelukkig', 'levenssystemen', 'ravenhurst', 'vrijheid', 'rinaldi', 'verachtelijk', 'nek', 'gedachte', 'kg,', 'vrijheid', 'vrijheid', 'ook', 'vrijheid'])
(['wij', 'zijn', 'louis', 'en', 'jessica', 'huang,', 'we', 'wonen', 'in', 'orlando'], ['verheffen', 'duitse', 'duitse', 'goedkoper', 'aggressive', 'gelukkig', 'gedachte', 'kg,', 'actrice', 'contract', 'is:', 'ravenhurst', 'taiko', 'machtiger', 'duitse'])
(['ik', 'ben', 'het', 'jaar', 'van', 'de', 'hond'], ['patrouille', 'patrouille', 'zinloos', 'bartolome', 'zeer', 'jongeman', 'produceer', 'intellect', 'vesting', 'vrijheid', 'vrijheid', 'vrijheid', 'verachtelijk', 'vrijheid', 'vrijheid'])

- RNN (1 layer): Since the pytorch implementation already has a specific RNN created using two linear modules, replacing them with an RNN module does not work.

- LSTM (1 layer): Since the pytorch implementation already has a specific RNN created using two linear modules, replacing them with an LSTM module does not work.

- RNN (2 layers): Since the pytorch implementation already has a specific RNN created using two linear modules, replacing them with an RNN module does not work.

- LSTM (2 layers): Since the pytorch implementation already has a specific RNN created using two linear modules, replacing them with an LSTM module does not work.

- Input sentences reversed (only training, not evaluation): 0.02149697704632642
(['we', 'staan', 'erop'], ['we', 'zijn', 'op', '<EOS>'])
(['dit', 'hadden', 'we', 'niet', 'verwacht'], ['we', 'zijn', 'op', '<EOS>'])
(['en', 'wij', 'staan', 'in', 'het', 'midden'], ['we', 'zijn', 'de', 'het', '<EOS>'])
(['je', 'bent', 'precies', 'wat', 'ik', 'nodig', 'heb'], ['je', 'bent', 'geen', '<EOS>'])
(['het', 'spijt', 'me,', 'noel'], ['ik', 'spijt', 'het', '<EOS>'])
(['je', 'bent', 'een', 'juweeltje'], ['je', 'bent', 'een', '<EOS>'])
(['graag', 'gedaan,', 'elizabeth'], ['je', 'bent', '<EOS>'])
(['kan', 'ik', 'aan,', 'geloof', 'me'], ['ik', 'ben', 'me', '<EOS>'])
(['wij', 'zijn', 'louis', 'en', 'jessica', 'huang,', 'we', 'wonen', 'in', 'orlando'], ['we', 'zijn', 'op', '<EOS>'])
(['ik', 'ben', 'het', 'jaar', 'van', 'de', 'hond'], ['ik', 'ben', 'de', 'van', '<EOS>'])

- Teacher forcing on (1.0): 0.046145425840054884
(['we', 'staan', 'erop'], ['we', 'zijn', 'allemaal', '<EOS>'])
(['dit', 'hadden', 'we', 'niet', 'verwacht'], ['we', 'zijn', 'hier', '<EOS>'])
(['en', 'wij', 'staan', 'in', 'het', 'midden'], ['wij', 'zijn', 'de', 'zoon', 'van', 'de', 'planeet', '<EOS>'])
(['je', 'bent', 'precies', 'wat', 'ik', 'nodig', 'heb'], ['je', 'bent', 'me', '<EOS>'])
(['het', 'spijt', 'me,', 'noel'], ['ik', 'ben', 'blij', 'dat', 'je', '<EOS>'])
(['je', 'bent', 'een', 'juweeltje'], ['je', 'bent', 'een', 'mens', '<EOS>'])
(['graag', 'gedaan,', 'elizabeth'], ['je', 'bent', 'geen', '<EOS>'])
(['kan', 'ik', 'aan,', 'geloof', 'me'], ['ik', 'ben', 'blij', 'dat', 'ik', 'je', 'me', '<EOS>'])
(['wij', 'zijn', 'louis', 'en', 'jessica', 'huang,', 'we', 'wonen', 'in', 'orlando'], ['we', 'zijn', 'in', 'het', 'in', 'het', 'leven', '<EOS>'])
(['ik', 'ben', 'het', 'jaar', 'van', 'de', 'hond'], ['ik', 'ben', 'de', 'zoon', 'van', 'de', 'zoon', '<EOS>'])

- Attention on: 0.025010259173115005
(['we', 'staan', 'erop'], ['we', 'zijn', 'het', '<EOS>'])
(['dit', 'hadden', 'we', 'niet', 'verwacht'], ['we', 'zijn', 'niet', 'op', '<EOS>'])
(['en', 'wij', 'staan', 'in', 'het', 'midden'], ['wij', 'zijn', 'de', '<EOS>'])
(['je', 'bent', 'precies', 'wat', 'ik', 'nodig', 'heb'], ['je', 'bent', 'je', '<EOS>'])
(['het', 'spijt', 'me,', 'noel'], ['ik', 'ben', 'me', '<EOS>'])
(['je', 'bent', 'een', 'juweeltje'], ['je', 'bent', 'een', '<EOS>'])
(['graag', 'gedaan,', 'elizabeth'], ['je', 'bent', 'je', '<EOS>'])
(['kan', 'ik', 'aan,', 'geloof', 'me'], ['ik', 'ben', '<EOS>'])
(['wij', 'zijn', 'louis', 'en', 'jessica', 'huang,', 'we', 'wonen', 'in', 'orlando'], ['we', 'zijn', 'op', 'op', 'op', '<EOS>'])
(['ik', 'ben', 'het', 'jaar', 'van', 'de', 'hond'], ['ik', 'ben', 'de', 'de', '<EOS>'])

**Comment by AlricRahl** [comment]
This issue has been attended to.
